openedx/features/idea/api_views.py

```diff
@@ -0,0 +1,26 @@
+from django.http import JsonResponse
+from rest_framework import status
+from rest_framework.authentication import SessionAuthentication, BasicAuthentication
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.views import APIView
+
+from openedx.features.idea.models import Idea
+
+
+class FavoriteAPIView(APIView):
+    """
+    FavoriteAPIView is used t toggle favorite idea for the user
+    """
+    authentication_classes = (SessionAuthentication, BasicAuthentication)
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request, idea_id):
+        response = dict(message='User is added to favorites')
+        http_status = status.HTTP_201_CREATED
+        user = request.user
+        idea = Idea.objects.get(pk=idea_id)
```

Comment:
How about we put a check here so that we nicely handle it if Idea doesn't exists.